### DISCUSSION TITLE
Using pagy_arel with a DISTINCT and a GROUP BY returning a weird pages number

### DISCUSSION THREAD

**Discussion by titanve** [ddnexus/pagy#477] [Q&A] [answered]
Hello,

I have an ActiveRecord query that has `.distinct` and `.group` and it is returning 10 items which is fine but I'm passing `items: 10` (`10` items per page) and the `pages` params that `pagy_arel` is return is `14` and the `count` is of `134` items, which is wrong and I wonder if `pagy_arel` (or `pagy` because it is happening with both) are taking in account the `.distinct`.

This is the query:
```
    def self.top_orders_slugs(query, brands_slugs:) # rubocop:disable Metrics/AbcSize
      query
        .where(brand_slug: brands_slugs)
        .select('product_brand_slug')
        .distinct
        .having(item_count/total_items_order.gt(0.8))
        .group('product_brand_slug', 'product_name_slug')
        .order('product_brand_slug ASC')
    end
```

And I use it like this in my controller:
```
query = Order.top_orders_slugs(Order.by_account(account), brands_slugs: @brands_data.keys)

pagy_arel(query, items: 10, page: orders_page_param)
```

Do I have to pass an extra param to `pagy_arel` to take in count the `.distinct`?

### Screenshots
![image](https://user-images.githubusercontent.com/4803351/222902264-71096c6f-0af4-4a60-a721-e1ea39b1ecaf.png)

Thank you.

**Comment by ddnexus** [comment, marked as answer]
I don't get what is wrong: from the screenshot: 137 pages means that you have 13 pages of 10 items (130 count so far) and the last page with 7 items, for a correct total  count of 137, and a correct total number of pages of 14.

What should it be instead?

Or you mean that the count in your data should be 134 and not 137?

In that case you might want to pass the right count. See [Custom count for custom scopes](https://ddnexus.github.io/pagy/docs/how-to/#custom-count-for-custom-scopes)

**Reply by titanve** [reply]
@ddnexus Hello,

I just have 10 items from the result of the query (before passing it to `pagy_arel`), so how come that just 10 items (having the number of items per page with `10`) generate `137` pages? It should be only `1` page.

Does `pagy_arel` and `pagy` work with `.distinct`? If not how may I make it work with it 🙏 

Thank you in advance.

**Reply by titanve** [reply]
@ddnexus 

It says that I should use `arel extra` which I guess I'm already using, right?
https://ddnexus.github.io/pagy/docs/how-to/#paginate-a-grouped-collection (the link to `arel extra` is broken)
![image](https://user-images.githubusercontent.com/4803351/222960611-50a8da9d-426a-4fd3-a5c3-e7810a0e527f.png)

**Comment by titanve** [comment]
@ddnexus Passing the `count` param did the trick but I wonder why do I have to pass it?

**Comment by ddnexus** [comment]
> I just have 10 items from the result of the query (before passing it to pagy_arel), so how come that just 10 items (having the number of items per page with 10) generate 137 pages? It should be only 1 page.

😱 You never said that you have 10 records in your query. You wrote 134 and 137.

> It says that I should use arel extra which I guess I'm already using, right?

You are the one that knows whether you required it or not. Please, read the arel doc. (BTW, `pagy_arel` in your code would error unless arel is loaded)

**Reply by titanve** [reply]
@ddnexus 

I'm using it and it is not returning an error, why would it return an error? I'm asking because I want to learn and understand more, if possible 🙏 

thanks!
